scheduler-worker/src/main/java/gov/nasa/jpl/aerie/scheduler/worker/ListenSchedulerCapability.java

```diff
@@ -27,38 +28,42 @@ public ListenSchedulerCapability(
     this.notificationQueue = notificationQueue;
   }
 
-  public void registerListener() throws SQLException {
-    PGConnection connection = this.dataSource.getConnection().unwrap(PGConnection.class);
-    connection.addNotificationListener(new SchedulingRequestNotificationListener());
+  public void registerListener() {
+    new Thread(() -> {
+      try (final var connection = this.dataSource.getConnection()) {
+        try (final var listenSimulationStatusAction = new ListenSchedulingRequestStatusAction(connection)) {
+          listenSimulationStatusAction.apply();
+        } catch (final SQLException ex) {
+          throw new DatabaseException("Failed to register as LISTEN to postgres database.", ex);
+        }
 
-    try (final var listenSchedulingRequestStatusAction = new ListenSchedulingRequestStatusAction(connection)) {
-       listenSchedulingRequestStatusAction.apply();
-    } catch (final SQLException ex) {
-      throw new DatabaseException("Failed to register as LISTEN to postgres database.", ex);
-    }
-  }
-
-  private final class SchedulingRequestNotificationListener implements PGNotificationListener {
-
-    @Override
-    public void notification(final int processId, final String channelName, final String payload){
-      logger.info("Received PSQL Notification: {}, {}, {}", processId, channelName, payload);
-      final var jsonValue = Json.createReader(new StringReader(payload)).readValue();
-      final var notificationPayload = postgresSchedulingRequestNotificationP
-          .parse(jsonValue)
-          .getSuccessOrThrow();
-      try {
-        notificationQueue.put(notificationPayload);
-      } catch (InterruptedException ex) {
-        // We are unable to handle this exception here so the thread interrupt flag is set to true
-        // so that code higher up the call stack can see that an interrupt was issued.
-        Thread.currentThread().interrupt();
+        while (true) {
```

Comment:
This loop should be `while (!Thread.currentThread().isInterrupted())`, as otherwise we're waiting on trying to put something into the `notificationQueue` in order to detect an interrupted signal.